src/App.vue

```diff
@@ -161,86 +301,232 @@ export default {
+            });
+        },
+
+        removeRow (row) {
+            this.inputColumn.splice(row, 1);
+        },
+
+        toggleModal (title, add, type) {
+            this.dialog.title = title;
+            this.dialog.dataType = type;
+            this.dialog.add = add;
+
+            add ? this.dialog.showDialogAdd = true : this.dialog.showDialogDelete = true;
+        },
+
+        closeModal () {
+            this.dialog.showDialogAdd = false;
+            this.dialog.showDialogDelete = false;
+            this.newName.name = null;
+        },
+
+        getElementsColumns () {
+            return _
+                .chain(this.inputColumn)
+                .transform((acc, input, index) => {
+                    let name = _.capitalize(input.name.toLowerCase());
+                    let type = input.type.toUpperCase();
+
+                    acc[index] = `${name} ${type}`;
+                })
+                .join(',')
+                .value();
+        },
+
+        executeQuery() {
+            let action, type, query, name;
+
+            action = this.dialog.add ? 'CREATE ' : 'DROP ';
+            type = this.dataTypeTable ? 'TABLE ' : 'DATABASE ';
+
+            this.newName.name
+                ? name = this.newName.name
+                : name = this.tableSelected.select || this.databaseSelected.select;
+
+            query = action + type + name;
+
+            if (this.newName.name) {
+                if (this.dataTypeTable) {
+                    query += ' ( ' + this.getElementsColumns() + ' )';
+                    this.sendQuery(query, ' table created!', 'Table already exists');
+                }
+                else {
+                    this.sendQuery(query, ' database created!', 'Database already exists');
+                }
+            }
+            else if (this.tableSelected.select || this.databaseSelected.select) {
+                if (this.dataTypeTable) {
+
+                    if (this.checkForeignKey()) {
+                        this.updateForeignKey(this.tableSelected.select);
+                    }
+                    else {
+                        this.sendQuery(query, ' table deleted!', 'Something went wrong.');
+                    }
+                }
+                else {
+                    this.sendQuery(query, ' database deleted!', 'Something went wrong.');
+                }
+            }
+        },
+
+        checkForeignKey () {
+            let query = "SELECT * FROM information_schema.KEY_COLUMN_USAGE WHERE TABLE_SCHEMA = '" + this.databaseActive + "'";
+
+            this.knex.raw(query).then(result => {
+                result = result[0];
+                result.forEach(line => {
+
+                    if (line.CONSTRAINT_NAME !== 'PRIMARY') {
+                        return true;
+                    }
+                    else {
+                        return false;
+                    }
+                })
+            })
+        },
+
+        updateForeignKey (table) {
```

Comment:
it's weird that you drop a table inside a method called `updateForeignKey`, try to make it more clear...